validity-primitive/src/Data/Validity/Primitive.hs

```diff
@@ -0,0 +1,16 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE CPP #-}
+
+module Data.Validity.Primitive where
+
+import Data.Validity
+
+import qualified Data.Primitive.Array as A
+
+#if MIN_VERSION_primitive(0,6,2)
+instance Validity a => Validity (A.Array a) where
+    validate = foldMap validate
+#else
+instance Validity a => Validity (A.Array a) where
+    validate = trivialValidation
```

Comment:
That sounds like a bad idea, why not validate . toList?

Reply:
Because in those old versions, there is no function "toList" or any equivalent, as far as I found.